### ISSUE TITLE
Add :binary source type?

### ISSUE THREAD

**Issue by lawik** [evadne/packmatic#12]
I used packmatic in a pre-existing application and I already have functions that get me the binary contents of a bunch of files. I don't particularly want to write them to a local file first and if wrapping them in a :stream is somehow useful, I think the library could do that.

I get that a `{:binary, "foo"}` is not particularly glamorous, though occasionally useful. But `{:dynamic, fn -> {:binary, get_binary_from_somewhere()} end}` was what I would have needed today. I ended up making a very simple one:

```
defmodule MyPackmaticBinary do
    @moduledoc """
    Implements a packmatic source for packaging straight
    from binaries.
    """
    @behaviour Packmatic.Source

    @impl true
    def validate(arg) do
      if is_binary(arg) do
        :ok
      else
        {:error, :not_binary}
      end
    end

    @impl true
    def init(arg) do
      {:ok, arg}
    end

    @impl true
    def read(arg) do
      case arg do
        nil -> :eof
        arg -> {arg, nil}
      end
    end
  end
```

**Comment by evadne** [comment]
Agreed

**Comment by evadne** [comment]
You could pass `{:stream, [binary]}` today and it should work as we use Enumerable for the stream source.

**Comment by evadne** [comment]
Addressed in v1.1.3 — use `{:source, enum/stream}` with IOList output
